numpy/__init__.pyi

```diff
@@ -1018,6 +1018,7 @@ def sort(
     axis: Optional[int] = ...,
     kind: Optional[_SortKind] = ...,
     order: Union[None, str, Sequence[str]] = ...,
+    by: Union[None, Sequence[ndarray], ndarray] = ...,
```

Comment:
```suggestion
    by: Optional[Sequence[ArrayLike]] = ...,
```
Currently only sequences (_i.e._ lists and tuples) of array-like objects are allowed, correct?

Reply:
Thanks, I am thinking if we should just make it `tuple` here. I don't mind making it sequences, but not sure there is much reason for not being strict (but I may be missing some earlier discussion).

Reply:
Makes sense, will make the change

Reply:
If we're going with tuple then `Optional[Tuple[ArrayLike, ...]]` should do the trick.